src/server/accessors/ModifyCreator.ts

```diff
@@ -72,4 +85,10 @@ export class ModifyCreator implements IModifyCreator {
 
         return this.bridges.getRoomBridge().create(result, builder.getMembersToBeAddedUsernames(), this.appId);
     }
+
+    private _finishUser(builder: IUserBuilder): Promise<string> {
+        const result = builder.getUser();
+        delete result.id;
+        return this.bridges.getUserBridge().create(result, this.appId);
```

Comment:
Almost wonder if we should do a check to ensure the user doesn't exist by the username, email, etc before we proceed to create the user that way we can gracefully handle the error instead of letting Rocket.Chat handle the error for us.

Reply:
Yes, @graywolf336 it would be better if we can check a user exists by username, email..etc but I couldn't find a way to check whether it exists from apps-engine. Anyways I think the server handles the case where a user exists gracefully throwing the reason for the error in logs.

Reply:
Then let's create it that way we can throw custom errors. The Apps Engine is an abstraction away from Rocket.Chat since the internals of Rocket.Chat can change at any point in time and we want the Apps Engine to be stable and consistent. Thus I would rather us do the extra work that way it is not a guessing game for App developers of what will happen and instead they can know up front. Does that make sense?